src/components/DirectusAsset.tsx

```diff
@@ -2,6 +2,21 @@ import { useContext, useEffect, useState } from 'react';
 import { DirectusAssetProps } from '@/types';
 import { DirectusContext } from '@/DirectusProvider';
 
+/**
```

Comment:
Is possibile to first better document components using @param @example etc. prior to introduce TSDoc? Or this is something only supported in the doc generation?

Reply:
Yes, TSDoc is in essence:
> TSDoc is a proposal to standardize the doc comments used in [TypeScript](http://www.typescriptlang.org/) code, so that different tools can extract content without getting confused by each other's markup. [TSDoc](https://tsdoc.org/).

So we can standardised doc comments to the code.

This PR introduces a way to automaticly generate Markdown documentation files based on those comments using [API-Extractor](https://api-extractor.com/).

Those files then reside in the docs directory and can be viewed directly in github or in any markdown viewer. They could also be hosted on GithubPages.

I think this would be a nicer and less error prone way then having all our docs in the README.